src/configuration/applier/hostescalation.cc

```diff
@@ -193,22 +163,16 @@ void applier::hostescalation::remove_object(
 
   // Find host escalation.
   std::string const& host_name{*obj.hosts().begin()};
-  uint64_t host_id{get_host_id(host_name)};
   std::pair<hostescalation_mmap::iterator, hostescalation_mmap::iterator> range{
       engine::hostescalation::hostescalations.equal_range(host_name)};
-  host_id_map::iterator hit{engine::host::hosts_by_id.find(host_id)};
+  host_map::iterator hit{engine::host::hosts.find(host_name)};
 
   for (hostescalation_mmap::iterator it{range.first}, end{range.second};
-       it != end;
-       ++it) {
-    std::list<std::shared_ptr<escalation> >& escalations(
-        hit->second->get_escalations());
-    for (std::list<std::shared_ptr<engine::escalation> >::iterator
-             itt{escalations.begin()},
-         next_itt{escalations.begin()},
-         end{escalations.end()};
-         itt != end;
-         itt = next_itt) {
+       it != end; ++it) {
+    std::list<escalation*>& escalations(hit->second->get_escalations());
+    for (std::list<engine::escalation*>::iterator itt{escalations.begin()},
+         next_itt{escalations.begin()}, end{escalations.end()};
```

Comment:
if you need the next iterator somewhere, you can use std::next

Reply:
You are right :-)